src/ethereum_test_forks/base_fork.py

```diff
@@ -0,0 +1,68 @@
+"""
+Abstract base class for Ethereum forks
+"""
+from typing import Type
+
+
+class BaseFork:
+    """
+    An abstract class representing an Ethereum fork.
+
+    Must contain all the methods used by every fork.
+    """
+
+    @classmethod
```

Comment:
Super nice and clean! I am being nit-picky as we don't yet do this anywhere else but we could add the following to further specify its an abstract class method:

```py
from abc import ABC, abstractmethod

class BaseFork(ABC):
    ...

    @abstractmethod
    def header_base_fee_required(...)
    ...
```

Reply:
Yea, strictly speaking, BaseFork should be abstract, and the definition of the starting behavior should happen at the first fork (Frontier), with the only caveat being that when we add a new resolver (like base_fee or excess_data_gas), it needs to be defined in BaseFork as an abstract function, and at Frontier with the starting implementation.